### PR TITLE
CHE-1849 Fix according to changes in Workspace API

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Fixes tests and other classes according to changes in Workspace API

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/1849

@evoevodin @sleshchenko please review

### CODE REVIEW

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 1]
```diff
@@ -124,7 +125,7 @@ public WorkspaceImpl startWorkspace(String workspaceId,
         return checkRamAndPropagateStart(workspace.getConfig(),
                                          envName,
                                          workspace.getNamespace(),
-                                         () -> super.startWorkspace(workspaceId, envName, accountId));
+                                         () -> super.startWorkspace(workspaceId, envName, accountId, null));
```
Comment (sleshchenko) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: why null? Should not be restore variable there?

**Comment by sleshchenko** [comment]
Should we remove "recoverWorkspace" here https://github.com/codenvy/codenvy/blob/ab81ba6da94485a21abe9fa0408bd7fa0eca0f0c/core/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/WorkspacePermissionsFilter.java#L82 ?

**Comment by sleshchenko** [comment]
other OK for me

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java (hunk 1 above)
Comment (vinokurig) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: done

**Comment by vinokurig** [comment]
Removed also interceptor bindings on 'recoverWorkspace'

File: core/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/WorkspacePermissionsFilter.java [hunk 2]
```diff
@@ -79,7 +79,6 @@ public void filter(GenericResourceMethod genericResourceMethod, Object[] argumen
                 workspaceId = ((String)arguments[0]);
                 action = DELETE;
                 break;
-            case "recoverWorkspace":
```
Comment (evoevodin) [inline core/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/WorkspacePermissionsFilter.java]: @sleshchenko is there any common mechanism which checks whether such methods exist?
I think this code must be covered by tests, as we usually have troubles in places like this after refactorings. 
@vinokurig consider adding such tests

Comment (sleshchenko) [inline core/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/WorkspacePermissionsFilter.java]: What about https://github.com/codenvy/codenvy/pull/448/files/5a4a3a9ff8769b39a48d21dabedac33cb45aa167#diff-15c913e41deb296c99e1a0ffcb8a5937L141 ?
Do you mean this mechanism?

Comment (evoevodin) [inline core/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/WorkspacePermissionsFilter.java]: I think this is pretty similar to interceptors compatibility tests like [this one](https://github.com/codenvy/customer-saas/blob/98fdeece736cebf6da844f3645b44cad1bbfa2b8/wsmaster/saas-marketo/src/test/java/com/codenvy/saas/marketo/UpdateProfileInterceptorTest.java). The main point here is that if intercepted class doesn't contain such method build MUST fail.

**Comment by evoevodin** [comment]
Ok

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**